### ISSUE TITLE
the code for the quantitative analysis

### ISSUE THREAD

**Issue by godchengzhihang** [shepnerd/inpainting_gmcnn#30]
Hello, is there a tensorflow version of the code for the quantitative analysis part of the paper?

**Comment by shepnerd** [comment]
No, there is not currently. The PSRN and SSIM values are measured by the built-in functions in matlab 2018. For the tensorflow version of psnr and ssim, you may refer to [https://www.tensorflow.org/api_docs/python/tf/image/psnr](https://www.tensorflow.org/api_docs/python/tf/image/psnr)  and [https://www.tensorflow.org/api_docs/python/tf/image/ssim](https://www.tensorflow.org/api_docs/python/tf/image/ssim). 
If you just wanna compute the psnr and ssim using python, you can refer to the following codes
`from skimage import measure`
Then,
`psnr = measure.compare_psnr(im1, im2)`
`ssim = measure.compare_ssim(im1, im2, multichannel=True)`.

**Comment by godchengzhihang** [comment]
Thank you very much!

**Comment by godchengzhihang** [comment]
First of all, thank you very much for your help. I still have a problem and I don't see the code for quantitative analysis in your code. How do you implement the code for quantitative analysis?

**Comment by shepnerd** [comment]
For imagenet and place2, we randomly pick n (200 for imagenet since msnps takes minutes to complete one image, and 2k for places2) images from the validation set and conduct inpainting on these images with the central rectangle mask. The psnr and ssim values are evaluated on the inpainting output and their corresponding ground truth. For paris streetview and celeba-hq, all validation/test images are used. The evaluations are also with central rectangle mask.

**Comment by godchengzhihang** [comment]
thank you for your help!

**Comment by godchengzhihang** [comment]
I am sorry to bother you again. Can you send me a reference for this part of the code? Because I really need this part of the code. @shepnerd

**Comment by shepnerd** [comment]
You can refer to [this](https://drive.google.com/file/d/10oBTOvF6sTAxvVGQl2d6SLBNF85dxQ2Z/view?usp=sharing).

If you cannot access to that file, you can email me for it.

**Comment by godchengzhihang** [comment]
thank you very much!
